src/gui/waveEdit.cpp

```diff
@@ -519,17 +526,71 @@ void FurnaceGUI::drawWaveEdit() {
                   if (waveGenScaleX<2) waveGenScaleX=2;
                   if (waveGenScaleX>256) waveGenScaleX=256;
                 }
+                ImGui::SetNextItemWidth(ImGui::GetContentRegionAvail().x);
+                if (CWSliderInt("##WGInterpolation", &waveInterpolation, 0, 3, waveInterpolations[waveInterpolation])) {
+                    if (waveInterpolation < 0) waveInterpolation = 0;
+                    if (waveInterpolation > 3) waveInterpolation = 3;
+                    //doGenerateWave();
+                }
                 ImGui::TableNextColumn();
                 if (ImGui::Button("Scale X")) {
-                  if (waveGenScaleX>0 && wave->len!=waveGenScaleX) e->lockEngine([this,wave]() {
-                    int origData[256];
-                    memcpy(origData,wave->data,wave->len*sizeof(int));
-                    for (int i=0; i<waveGenScaleX; i++) {
-                      wave->data[i]=origData[i*wave->len/waveGenScaleX];
-                    }
-                    wave->len=waveGenScaleX;
-                    MARK_MODIFIED;
-                  });
+                    if (waveGenScaleX > 0 && wave->len != waveGenScaleX) e->lockEngine([this, wave]() {
+                        int origData[256];
+                        // Copy original wave to temp buffer
+                        assert(wave->len <= 256);
```

Comment:
Don't assert. We don't want the program to crash.

Reply:
Fixed the assertion issue (replaced with showError() and fixed a small bug with the interpolation.